sys/include/net/gnrc/ppp/fsm.h

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright (C) 2016 José Ignacio Alamos <jialamos@uc.cl>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_gnrc_ppp Option Negotiation Automaton FSM header definitions
+ * @ingroup     net_gnrc_ppp
+ * @{
+ *
+ * @file
+ * @brief  Option Negotiation Automaton FSM definitions
+ *
+ * @author  José Ignacio Alamos <jialamos@uc.cl>
+ */
+#ifndef PPP_FSM_H_
+#define PPP_FSM_H_
+
+#include "net/gnrc/ppp/opt.h"
+#include "net/gnrc/ppp/prot.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/*Function flags*/
+#define F_TLU (1U << 0)
+#define F_TLD (1U << 1)
+#define F_TLS (1U << 2)
+#define F_TLF (1U << 3)
+#define F_IRC (1U << 4)
+#define F_ZRC (1U << 5)
+#define F_SCR (1U << 6)
+#define F_SCA (1U << 7)
+#define F_SCN (1U << 8)
+#define F_STR (1U << 9)
+#define F_STA (1U << 10)
+#define F_SCJ (1U << 11)
+#define F_SER (1U << 12)
+
+/* flags for setting FSM codes */
+#define FLAG_CONF_REQ (1 << 0)
+#define FLAG_CONF_ACK (1 << 1)
+#define FLAG_CONF_NAK (1 << 2)
+#define FLAG_CONF_REJ (1 << 3)
+#define FLAG_TERM_REQ (1 << 4)
+#define FLAG_TERM_ACK (1 << 5)
+#define FLAG_CODE_REJ (1 << 6)
+#define FLAG_PROT_REJ (1 << 7)
+#define FLAG_ECHO_REQ (1 << 8)
+#define FLAG_ECHO_REP (1 << 9)
+#define FLAG_DISC_REQ (1 << 10)
+#define FLAG_IDENT (1 << 11)
+#define FLAG_TIME_REM (1 << 12)
+
+
+#define PPP_MAX_TERMINATE (3)       /**< Maximum number of Terminate Request retransmission*/
+#define PPP_MAX_CONFIG (10)         /**< Maximum number of Configure Request retransmission */
+
+#define OPT_PAYLOAD_BUF_SIZE (100)  /**< Size of PPP packet payload (only for Auth, LCP, and NCP packets) */
+
+#define OPT_ENABLED (1)             /**< FSM option enabled */
+#define OPT_REQUIRED (2)            /**< FSM option is required */
+/**
+ * @brief Option Negotiation Automaton FSM events
+ *
+ * @see https://tools.ietf.org/html/rfc1661#section-4.3
+ */
+typedef enum {
+    E_UP,
+    E_DOWN,
+    E_OPEN,
+    E_CLOSE,
+    E_TOp,
+    E_TOm,
+    E_RCRp,
+    E_RCRm,
+    E_RCA,
+    E_RCN,
+    E_RTR,
+    E_RTA,
+    E_RUC,
+    E_RXJp,
+    E_RXJm,
+    E_RXR,
+    PPP_NUM_EVENTS
+} fsm_event_t;
+
+
+/**
+ * @brief Option Negotiation Automaton FSM states
+ *
+ * @see https://tools.ietf.org/html/rfc1661#section-4.2
+ */
+typedef enum {
+    S_UNDEF = -1,
+    S_INITIAL = 0,
+    S_STARTING,
+    S_CLOSED,
+    S_STOPPED,
+    S_CLOSING,
+    S_STOPPING,
+    S_REQ_SENT,
+    S_ACK_RCVD,
+    S_ACK_SENT,
+    S_OPENED,
+    PPP_NUM_STATES
+} fsm_state_t;
+
+
+/**
+ * @brief Option Negotiation Automaton FSM transitions
+ *
+ * @see https://tools.ietf.org/html/rfc1661#section-4.1
+ */
+static const int8_t state_trans[PPP_NUM_EVENTS][PPP_NUM_STATES] = {
+    { S_CLOSED, S_REQ_SENT, S_UNDEF, S_UNDEF, S_UNDEF, S_UNDEF, S_UNDEF, S_UNDEF, S_UNDEF, S_UNDEF },
```

Comment:
Here and below: [Line-length](https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions#general)